### DISCUSSION TITLE
Sonoff TH10 does not display Temperatur Value

### DISCUSSION THREAD

**Discussion by haweneu** [arendst/Tasmota#22346] [Q&A]
Hi all,
i programmed an older Sonoff TH10 (sensor DS18B20) with Tasmota 14.3.0. After complete programming it does not display the temperature in the upper right corner. With the older Tasmota 9.5.0 ist works. Both versions have the same settings. Difference ist in GPIO14 name of the temperature sensor. Tasmota 14.3.0 shows "DS18x20" and right of this text a selection box with values 1 to 4. I tested all 4 values! Tasmota 9.5.0 shows only  "DS18x20".

dont kno why!

please help me.

best regards
Werner from germany

**Comment by sfromis** [comment]
The dropdown to the right with newer Tasmota versions is due to a new feature of supporting DS18x20 on up to 4 input gpios, instead of only one. Stick with the default index of 1 unless you are using more than one gpio.

Did you enable `SetOption74` to enable internal pull-up?

**Reply by haweneu** [reply]
Hi,
Thanks for the Info. I tested the dropout with all 4 Versions! No value Display.
I never used  setoption74! With the older Tasmota Version it works without this!
But i will test it with this Option and Report my result!

Best regards
Werner

**Reply by haweneu** [reply]
Hi,
today i installed again the newest version tasmota release 14.3.0 and it works!
very strange!!! yesterday i installed the same version many times and never found a temperatur display!

I didnt use setoption 74!

But for security reasons  i will use this in the future!
if a pullup is needed then it must be activated!

thanks for you help.

**Comment by lalo-uy** [comment]
The pullup of 3 to 5k is needed for the 18B20. If the sensor work the
pullup is in the hardware, no need to enable an internal one that surely is
10k or more.

**Reply by sfromis** [reply]
However, with ESP8266, the weak internal pullup *usually* works fine for a single DS18x20, no external pullup. Still, in some cases circumstances around connections, wiring, etc, may result in external pullup being needed. The important thing is to *always* have pullup, internal or external.

For context, with ESP32, external pullup is always needed for Tasmota, but this thread was about Sonoff TH10 having ESP8266.

**Reply by haweneu** [reply]
Hi,
Ok. I understood!
Thanks for your help.
